src/map/script.cpp

```diff
@@ -25462,11 +25462,29 @@ BUILDIN_FUNC(getenchantgrade){
 		return SCRIPT_CMD_FAILURE;
 	}
 
-	if( current_equip_item_index == -1 ){
+	int i, num;
+
+	if (script_hasdata(st, 2))
+		num = script_getnum(st, 2);
+	else
+		num = EQI_COMPOUND_ON;
+
+	if (num == EQI_COMPOUND_ON)
+		i = current_equip_item_index;
+	else if (equip_index_check(num))
+		i = pc_checkequip(sd, equip_bitmask[num]);
+	else {
+		ShowError( "buildin_getenchantgrade: Unknown equip index '%d'\n", num );
+		script_pushint(st,-1);
 		return SCRIPT_CMD_FAILURE;
 	}
 
-	script_pushint( st, sd->inventory.u.items_inventory[current_equip_item_index].enchantgrade );
+	item* itm = &sd->inventory.u.items_inventory[i];
```

Comment:
Here you already access invalid index already. So here it might crash if i < 0 or i >= MAX_INVENTORY

Reply:
wait I have an excuse, it was late !